src/app/dev/DevToys.Tools/Helpers/XmlHelper.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System.Text;
+using System.Text.RegularExpressions;
+using DevToys.Tools.Models;
+using System.Xml;
+using Microsoft.Extensions.Logging;
+using DevToys.Api.Core;
+
+namespace DevToys.Tools.Helpers;
+
+internal static class XmlHelper
+{
+    /// <summary>
+    /// Detects whether the given string is a valid Xml or not.
+    /// </summary>
+    internal static bool IsValid(string? input, ILogger logger)
+    {
+        if (string.IsNullOrWhiteSpace(input))
+        {
+            return false;
+        }
+
+        input = input!.Trim();
+
+        if (!input.StartsWith("<") || !input.EndsWith(">"))
+        {
+            return false;
+        }
+
+        try
+        {
+            var xmlDocument = new XmlDocument();
+
+            // If loading failed, it's not valid Xml.
+            xmlDocument.LoadXml(input);
+
+            return true;
+        }
+        catch (XmlException)
+        {
+            return false;
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, "Invalid data detected 'input'", input);
+            return false;
+        }
+    }
+
+    /// <summary>
+    /// Format a string to the specified Xml format.
+    /// </summary>
+    internal static ResultInfo<string> Format(string? input, Indentation indentationMode, bool newLineOnAttributes, ILogger logger)
+    {
+        if (string.IsNullOrWhiteSpace(input))
+        {
+            return new(string.Empty, false);
+        }
+
+        input = input!.Trim();
```

Comment:
Is the Trim necessary here? Would `XmlDocument` fail to load if there's a space at the beginnig/end of the string?

Reply:
It is not necessary. I tested with a string containing spaces at the beginning and end, and the `XmlDocument` correctly removed them as part of the formatting.